flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/v2/FilesCommitter.java

```diff
@@ -0,0 +1,224 @@
+          flinkJobId = commitRequest.getCommittable().jobID();
+        } catch (Exception e) {
+          commitRequest.signalFailedWithUnknownReason(e);
+        }
+      }
+
+      commitOperation(appendFiles, dataFilesNum, 0, "append", flinkJobId);
+      writeResults.forEach(CommitRequest::signalAlreadyCommitted);
+    } else {
+      // To be compatible with iceberg format V2.
+      RowDelta rowDelta = table.newRowDelta().scanManifestsWith(workerPool);
+      int numDataFiles = 0;
+      int numDeleteFiles = 0;
+      String flinkJobId = "";
+      for (CommitRequest<FilesCommittable> commitRequest : writeResults) {
+        try {
+          WriteResult result = commitRequest.getCommittable().committable();
+          // We don't commit the merged result into a single transaction because for the sequential transaction txn1 and
+          // txn2, the equality-delete files of txn2 are required to be applied to data files from txn1. Committing the
+          // merged one will lead to the incorrect delete semantic.
+
+          // Row delta validations are not needed for streaming changes that write equality deletes. Equality deletes
+          // are applied to data in all previous sequence numbers, so retries may push deletes further in the future,
+          // but do not affect correctness. Position deletes committed to the table in this path are used only to delete
+          // rows from data files that are being added in this commit. There is no way for data files added along with
+          // the delete files to be concurrently removed, so there is no need to validate the files referenced by the
+          // position delete files that are being committed.
+
+          numDataFiles = numDataFiles + result.dataFiles().length;
+          Arrays.stream(result.dataFiles()).forEach(rowDelta::addRows);
+
+          numDeleteFiles = numDeleteFiles + result.deleteFiles().length;
+          Arrays.stream(result.deleteFiles()).forEach(rowDelta::addDeletes);
+
+          flinkJobId = commitRequest.getCommittable().jobID();
+        } catch (Exception e) {
+          commitRequest.signalFailedWithUnknownReason(e);
+        }
+      }
+
+      commitOperation(rowDelta, numDataFiles, numDeleteFiles, "rowDelta", flinkJobId);
+      writeResults.forEach(CommitRequest::signalAlreadyCommitted);
+    }
+  }
+
+  private void commitOperation(SnapshotUpdate<?> operation, int numDataFiles, int numDeleteFiles, String description,
+                               String newFlinkJobId) {
+    LOG.info("Committing {} with {} data files and {} delete files to table {}", description, numDataFiles,
+        numDeleteFiles, table);
+    snapshotProperties.forEach(operation::set);
+    // custom snapshot metadata properties will be overridden if they conflict with internal ones used by the sink.
+    operation.set(FLINK_JOB_ID, newFlinkJobId);
+
+    long start = System.currentTimeMillis();
+    operation.commit(); // abort is automatically called if this fails.
```

Comment:
Maybe we should handle `CommitStateUnknownException` differently, as in this case we do not know if the commit was successful, or not. If we retry to commit in this case we might end up with duplicated data.